src/Core/Utility.re

```diff
@@ -263,6 +263,12 @@ module Option = {
     | Some(x) => f(x)
     | None => None;
 
+  let flatten =
+    fun
+    | Some(Some(x)) => Some(x)
+    | Some(None) => None
+    | None => None;
```

Comment:
This is equivalent to 
```
    | Some(x) => x
    | None => None;
```

Because the last branch is `None`, `x` has to be an `option(_)`.

Reply:
Makes sense - nice!